src/arch/ia32.rs

```diff
@@ -1,5 +1,6 @@
 use core::arch::asm;
 
+pub mod address;
```

Comment:
Is this even compiling for ia32 target ?

I think you should move the address file from ia32e to ia32, keep this module. But instead of creating a submodule for ia32e, you should probably reexport the module from ia32 (pub use).